electrum/commands.py

```diff
@@ -1322,6 +1322,46 @@ async def reverse_swap(self, lightning_amount, onchain_amount, wallet: Abstract_
             'onchain_amount': format_satoshis(onchain_amount_sat),
         }
 
+    @command('n')
+    async def convert_currency(self, from_amount=1, from_ccy = None, to_ccy = None):
+        """Converts the given amount of currency to another using the
+        configured exchange rate source.
+        """
+        if not self.daemon.fx.is_enabled():
+            raise Exception("FX is disabled. To enable, run: 'electrum setconfig use_exchange_rate true'")
+        # Default currencies
+        if from_ccy is None and to_ccy is None:
+            from_ccy = 'BTC'
+            to_ccy = self.daemon.fx.ccy
+        elif from_ccy is None:
+            from_ccy = 'BTC'
+        elif to_ccy is None:
+            to_ccy = 'BTC'
```

Comment:
The defaults should behave in a way such that if one or both of `from_ccy` and `to_ccy` are unset, they should be set to the `(fx.ccy, "BTC")` pair (or its reverse). The current code does not satisfy that:
```
$ ./run_electrum --testnet convert_currency --from_amount 1 --from_ccy BTC
{
    "from_amount": "1.00000000",
    "from_ccy": "BTC",
    "source": "CoinGecko",
    "to_amount": "1.00000000",
    "to_ccy": "BTC"
}
```
-----
Try e.g.:
```suggestion
        if from_ccy is None:
            from_ccy = "BTC" if to_ccy != "BTC" else self.daemon.fx.ccy
        if to_ccy is None:
            to_ccy = "BTC" if from_ccy != "BTC" else self.daemon.fx.ccy
```
With that, I get the following, which looks ok:
```
$ ./run_electrum convert_currency --from_amount 2
{
    "from_amount": "2.00000000",
    "from_ccy": "BTC",
    "source": "CoinGecko",
    "to_amount": "32472.72",
    "to_ccy": "EUR"
}
$ ./run_electrum convert_currency --from_amount 2 --from_ccy BTC
{
    "from_amount": "2.00000000",
    "from_ccy": "BTC",
    "source": "CoinGecko",
    "to_amount": "32472.72",
    "to_ccy": "EUR"
}
$ ./run_electrum convert_currency --from_amount 1000 --from_ccy EUR
{
    "from_amount": "1000.00",
    "from_ccy": "EUR",
    "source": "CoinGecko",
    "to_amount": "0.06159017",
    "to_ccy": "BTC"
}
$ ./run_electrum convert_currency --from_amount 1000 --to_ccy BTC
{
    "from_amount": "1000.00",
    "from_ccy": "EUR",
    "source": "CoinGecko",
    "to_amount": "0.06159017",
    "to_ccy": "BTC"
}
$ ./run_electrum convert_currency --from_amount 2 --to_ccy EUR
{
    "from_amount": "2.00000000",
    "from_ccy": "BTC",
    "source": "CoinGecko",
    "to_amount": "32472.72",
    "to_ccy": "EUR"
}
```
Feel free to write unit tests if you like, e.g. extend `tests/test_commands.py`.

Reply:
> The defaults should behave in a way such that if one or both of `from_ccy` and `to_ccy` are unset, they should be set to the `(fx.ccy, "BTC")` pair (or its reverse).

Well, it's a bit more complicated, because of the case where the caller specifies exactly one of the ccy args to a third value, e.g. GBP. In that case I think the best behaviour is to default the other ccy to BTC.